test/backends/xeon/run_test.py

```diff
@@ -0,0 +1,112 @@
+            for line in p.stdout.readlines():
+                segs = str(line, "utf-8").strip().split("-")
+                if segs[-1].strip() == "Use JeMalloc memory allocator":
+                    num += 1
+        assert num > 0, "Failed to use jemalloc"
+
+    def test_tcmalloc(self):
+        num = 0
+        with subprocess.Popen(f"python -m torch.backends.xeon.run_cpu --ninstances 1 --enable_tcmalloc --log_path \
+            {self._test_dir} --disable_numactl --no_python pwd",
+                              shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT) as p:
+            for line in p.stdout.readlines():
+                segs = str(line, "utf-8").strip().split("-")
+                if segs[-1].strip() == "Use TCMalloc memory allocator":
+                    num += 1
+        assert num > 0, "Failed to use tcmalloc"
+
+    def test_iomp(self):
+        num = 0
+        with subprocess.Popen(f"python -m torch.backends.xeon.run_cpu --ninstances 1 --use_default_allocator \
+            --log_path {self._test_dir} --disable_numactl --no_python pwd",
+                              shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT) as p:
+            for line in p.stdout.readlines():
+                segs = str(line, "utf-8").strip().split("-")
+                if segs[-1].strip() == "Using Intel OpenMP":
+                    num += 1
+        assert num > 0, "Failed to use iomp"
+
+    def test_default_config(self):
+        num = 0
+        with subprocess.Popen(f"python -m torch.backends.xeon.run_cpu --ninstances 1 --use_default_allocator \
+            --disable_iomp --log_path {self._test_dir} --disable_numactl --no_python pwd",
+                              shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT) as p:
+            for line in p.stdout.readlines():
+                segs = str(line, "utf-8").strip().split("-")
+                if segs[-1].strip() == "Use JeMalloc memory allocator":
+                    num += 1
+                if segs[-1].strip() == "Use TCMalloc memory allocator":
+                    num += 1
+                if segs[-1].strip() == "Using Intel OpenMP":
+                    num += 1
+        assert num == 0, "Failed to use default config"
+
+    def test_multi_threads(self):
+        num = 0
+        with subprocess.Popen(f"python -m torch.backends.xeon.run_cpu --ninstances 4 --use_default_allocator \
+            --disable_iomp --disable_numactl --log_path {self._test_dir} --no_python pwd",
+                              shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT) as p:
+            for line in p.stdout.readlines():
+                segs = str(line, "utf-8").strip().split("-")
+                if segs[-1].strip() == "pwd":
+                    num += 1
+        assert num == 4, "Failed to launch multiple instances for inference"
+
+if __name__ == "__main__":
+    subprocess.call("yes | conda install -c conda-forge gperftools jemalloc", shell=True)
```

Comment:
I don't think we ever do such things in our tests.
I think it is better to raise a nice error if they are not available and hint the user towards installing these.

Reply:
This command is to enable testing of preloading jemalloc and tcmalloc. If this cannot be executed, then I'll need to remove `test_jemalloc` (line 51) and `test_tcmalloc` (line 62) from this UT script. Is it OK to remove these 2 test functions?

Reply:
I think the way to go would be to check if these are installed and if not, either:
- Skip the tests that require it.
- Raise an error asking the user to install these before running the test.